src/components/shared/levelSelect.tsx

```diff
@@ -1,44 +1,62 @@
+
+import {useState, useEffect} from 'react';
 import '../../styles/levelSelect.scss';
-import {useState} from 'react';
 import Dropdown, { Option } from 'react-dropdown';
-import { BsArrowUpShort, BsArrowDownShort } from 'react-icons/bs';
+import { BsCaretLeftFill , BsCaretRightFill, BsFillCaretUpFill, BsFillCaretDownFill } from 'react-icons/bs';
+import { useHistory } from 'react-router-dom';
+
+interface LevelSelectProps {
+  pageOptions: string[];
+  currentPage: string;
+}
 
+function LevelSelect(props:LevelSelectProps): JSX.Element {
+  const { pageOptions, currentPage } = props;
 
-function LevelSelect(): JSX.Element {
+  const [index, setIndex] = useState(pageOptions.indexOf(currentPage));
 
-  const pages = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const [reloadTime, setReloadTime] = useState(Date.now());
 
-  const [currentPage, setCurrentPage] = useState(6); // change this to current page once set up
+  const history = useHistory();
 
   const createDropDownItems = ():Option[] => {
-    return pages.map((item) => {
+    return pageOptions.map((item, itemIndex) => {
       return  ({
-        label: <div>{item}</div>,
+        label: <div>{itemIndex + 1}</div>,
         value: item,
       });
     });
   };
 
-  const pageArray = createDropDownItems();
-
-  const onPageChange = (currentOption:Option):void => {
-    // add functionality here to change the current page
-    // the selected option is passed in automatically
-    setCurrentPage(parseInt(currentOption.value));
+  const onLevelChange = (newIndex:number) => {
+    setReloadTime(Date.now());
+    setIndex(newIndex);
   };
 
+  useEffect(() => {
+    if (index > 8) {
```

Comment:
Page options. Length

Reply:
Component should be dependent on input and not specific for one use case <unless otherwise noted>